landfill/landfill.py

```diff
@@ -0,0 +1,160 @@
+__all__ = [
+    "fourier_encode",
+    "prep_data",
+    "create_model",
+    "backfill",
+]
+import numpy as np
+from tensorflow import keras as tfk
+from matplotlib import pyplot as plt
+
+###############################################################################
+# This is just a little example using the the keras functional API and        #
+# fourier features.                                                           #
+###############################################################################
+
+
+# n_features = 100 #Number of fourier features
+# sigma=30.
+
+
+# Encode fourier features
+def fourier_encode(im, sigma, n_features):
+    """
+    Parameters
+    ----------
+    im : (M, N) array-like
+    sigma : float
+        The scale of the Gaussian from which the weights are drawn
+    n_features : int
+        How many fourier features to sample
+
+    Returns
+    -------
+    gamma : () array-like
+        The encoded image
+    """
+    idx = np.indices(im.shape).reshape((2, -1)).T
+    V = idx / idx.max()
+    B = np.random.normal(scale=sigma, size=(2, n_features))
+    return np.hstack((np.cos(2 * np.pi * V @ B), np.sin(2 * np.pi * V @ B)))
+
+
+def prep_data(im, mask, sigma, n_features=20):
+    """
+    Parameters
+    ----------
+    im : (M, N) array-like
+    mask : (M, N) array-like of bool
+    sigma : float
+        The scale of the Gaussian from which the weights are drawn
+    n_features : int, default: 20
+
+    Returns
+    -------
+    X, y : array-like
+        The training data for the model
+    gamma
+        The encoded image
+    """
+    gamma = fourier_encode(im, sigma=sigma, n_features=n_features)
+    y = im.flatten().astype(np.float32)
+
+    # Removed masked pixels
+    y = y[~mask.flatten()]
+    X = gamma[~mask.flatten()].astype(np.float32)
+    return X, y, gamma
+
+
+def create_model(
+    n_features, n_layers=20, hidden_width=3, loss=tfk.losses.Poisson(), optimizer="adam"
```

Comment:
i don't think it is "safe" to pass `loss=tfk.losses.Poisson()`, because it is not immutable. Although maybe that is an oldschool thing? I think this is a use case for `loss=None` followed by some control flow in the function body.

Reply:
also... saying not immutable is obvious silly.